config/processors/syslog_log_security_sdwan.app.conf

```diff
@@ -102,197 +114,263 @@ filter {
 
-    if [tmp][dataset] == "alarmlog" {
-      mutate {
-        rename => { "[tmp][rest_msg][appliancename]" => "[host][hostname]" }
-        rename => { "[tmp][rest_msg][tenantname]" => "[organization][name]"}
-        rename => { "[tmp][rest_msg][alarmtype]" => "[error][type]"}
-        rename => { "[tmp][rest_msg][alarmcause]" => "[event][action]"}
-        rename => { "[tmp][rest_msg][alarmseqno]" => "[event][id]"}
-        rename => { "[tmp][rest_msg][alarmtext]" => "[error][message]"}
-        rename => { "[tmp][rest_msg][serialnum]" => "[host][id]"}
-
-        add_field => { "[rule][category]" => "%{[[tmp][rest_msg][alarmclass]]}"}
-        add_field => { "event.reason" => "%{[[tmp][rest_msg][alarmkind]]}"}
-        add_field => { "[event][type]" => "%{[[tmp][rest_msg][alarmeventtype]]}"}
-        add_field => { "[log][syslog][severity][name]" => "%{[[tmp][rest_msg][alarmseverity]]}"}
-        add_field => { "[event][code]" => "%{[[tmp][rest_msg][alarmkey]]}"}
-        
-      }
-      date {
-        match => [ "[tmp][rest_msg][generatetime]", "UNIX" ]
-        target => "[event][start]"
-      }
+  if [@metadata][tmp][dataset] == "alarmlog" {
 
-      translate {
-        source => "[event][action]"
-        target => "[event][outcome]"
-        dictionary => {
-          "performancedegradeed" => "failure"
-          "outofservice" => "failure"
-        }
-        fallback => "unknown"
+    mutate {
+      add_field => { "[event][reason]" => "unknown" }
+    }
+
+    mutate {
+      rename => { "[@metadata][tmp][rest_msg][appliancename]" => "[host][hostname]" } # the appliance sending logs is not the appliance where the event took place. this is provided by appliancename
+      rename => { "[@metadata][tmp][rest_msg][tenantname]" => "[organization][name]"}
+      rename => { "[@metadata][tmp][rest_msg][alarmtype]" => "[error][type]"}
+      rename => { "[@metadata][tmp][rest_msg][alarmseqno]" => "[event][sequence]"}
+      rename => { "[@metadata][tmp][rest_msg][alarmtext]" => "[error][message]"}
+      rename => { "[@metadata][tmp][rest_msg][serialnum]" => "[host][id]"}
+      rename => { "[@metadata][tmp][rest_msg][alarmeventtype]" => "[error][code]"}
+      update => { "[event][reason]" => "%{[@metadata][tmp][rest_msg][alarmcause]}"}
+      tag_on_failure => "_mutate_error_alarmlog"
+    }
+
+    mutate {
+      rename => {"[tags]" => "[@metadata][tmp][original_tags]"}
+    }
+
+    mutate {
+      add_tag => [
+        "sdwan_alarm_class=%{[@metadata][tmp][rest_msg][alarmclass]}", 
+        "sdwan_alarm_kind=%{[@metadata][tmp][rest_msg][alarmkind]}", 
+        "sdwan_alarm_severity=%{[@metadata][tmp][rest_msg][alarmseverity]}",
+        "sdwan_alarm_key=%{[@metadata][tmp][rest_msg][alarmkey]}",
+        "sdwan_alarm_cause=%{[@metadata][tmp][rest_msg][alarmcause]}",
+        "sdwan_alarm_onwer=%{[@metadata][tmp][rest_msg][alarmowner]}"
+        ]
+    }
+
+    mutate {
+      rename => {"[tags]" => "[labels][all]"}
```

Comment:
@tbies, will you be searching or using this "labels.all" field? If yes, we need to change this to ecs field.

Reply:
FYI the reason i put the versa specific fields into [labels][all] was because some of these fields have no place under ECS but they might be used be teams looking into sdwan logs. This is in alignment with https://www.elastic.co/guide/en/ecs/current/ecs-custom-fields-in-ecs.html